### PR TITLE
Introduce logOwnerReasons option

### PR DESCRIPTION
**Pull request by Hypnosphi**
See #99 for details

TODO:
* [x] Tests
* [x] Docs

### CODE REVIEW

**Comment by vzaidman** [comment]
beautiful!

File: src/whyDidYouRender.js [hunk 1]
```diff
@@ -90,6 +93,18 @@ export default function whyDidYouRender(React, userOptions){
   const origCreateFactory = React.createFactory
 
   let componentsMap = new WeakMap()
+  const ownerDataMap = new WeakMap()
+  const hooksRef = {current: []}
+
+  Object.defineProperty(React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, 'current', {
```
Comment (vzaidman) [inline src/whyDidYouRender.js:99]: let's use a more definite name for this one with WDYR prefix

Comment (Hypnosphi) [reply src/whyDidYouRender.js:99]: This is what React itself uses to track current owner, I just intercept the writes

Comment (vzaidman) [reply src/whyDidYouRender.js:99]: oh nice. i got it. please add a comment about it on top of it.
something like
```
// intercepting assigns to ReactCurrentOwner.current
```

**Comment by Hypnosphi** [comment]
I think it's ready for review

**Comment by vzaidman** [comment]
hey i added a small commit and it conflicts, please take a look

**Comment by vzaidman** [comment]
Thank you!
I'm releasing this beautiful feature as alpha!

**Comment by vzaidman** [comment]
@Hypnosphi released as "4.1.0-alpha.0"
I'm going to work with it a little before releasing 4.1.0.
Thanks again!

**Comment by vzaidman** [comment]
@Hypnosphi hey!
did you use it in your code?
did you see any problems with it?
it worked just fine for me!

**Comment by vzaidman** [comment]
released in
https://github.com/welldone-software/why-did-you-render/releases/tag/v4.1.1!

**Comment by Hypnosphi** [comment]
Works well for my usecases

**Pull request merged**